src/constraint/encoding.ts

```diff
@@ -138,8 +139,30 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       }
       return true;
     }
-  },
-  {
+  },{
+    name: 'scalePropertiesSupportedByScaleType',
+    description: 'Scale properties must be supported by correct scale type',
+    properties: [Property.SCALE, Property.SCALE_TYPE, Property.SCALE_ZERO], // Add more scale properties!
+    allowEnumSpecForProperties: false,
+    strict: true,
+    satisfy: (encQ: EncodingQuery, schema: Schema, opt: QueryConfig) => {
+      if (encQ.scale) {
+        const scale: ScaleQuery = encQ.scale as ScaleQuery;
+
+        for (let scaleProp in SUPPORTED_SCALE_PROPERTY_INDEX) {
```

Comment:
Instead of looping over all possible scale properties, looping over existed properties of `encQ.scales` might be faster?